### PR TITLE
Add cloud cache support for aml models

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

Add cloud cache support for aml models

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: olive/engine/cloud_cache_helper.py
Comment (jambayk) [inline olive/engine/cloud_cache_helper.py:1]: this PR supports aml models when running from local workflow host. However, it still doesn't support aml models when running on an aml workflow host. To the aml host, the model appears as a local model so some more bookkeeping would be required to inform it that it is an aml model. 

Will this be implemented soon? As a follow-up PR of part of this?

File: olive/engine/engine.py [hunk 1]
```diff
@@ -803,21 +802,15 @@ def _run_passes(
...
-            if not (
-                model_config.type.lower() == "hfmodel"
-                and create_resource_path(model_config.config.get("model_path")).is_string_name()
-            ):
+            if not is_valid_cloud_cache_model(model_config):
                 logger.warning(
-                    "Only HfModel with huggingface id as model_path is supported by cloud cache. Setting"
-                    " enable_cloud_cache=False."
+                    "Only HfModel with huggingface id as model_path "
+                    "or HFModel from Azure ML registry model is supported by cloud cache. "
```
Comment (devang-ml) [inline olive/engine/engine.py:808]: - HfModel vs HFModel
- HFModel is Olive concept that user may need to be familiar with.

File: olive/engine/cloud_cache_helper.py
Comment (xiaoyu-work) [reply olive/engine/cloud_cache_helper.py:1]: Yeah i'm still looking into aml models with aml workflow host.

File: olive/engine/cloud_cache_helper.py [hunk 2]
```diff
@@ -92,22 +99,20 @@ def get_hash_key(
...
         hf_hub_model_commit_id = None
         model_config_copy = deepcopy(model_config)
-        if (
-            input_model_hash is None
-            and model_config.type.lower() == "hfmodel"
-            and create_resource_path(model_config.config["model_path"]).is_string_name()
-        ):
-            hf_hub_model_commit_id = repo_info(model_config.config["model_path"]).sha
+        model_path_resource_path = create_resource_path(model_config.config.get("model_path"))
+        if input_model_hash is None:
```
Comment (jambayk) [inline olive/engine/cloud_cache_helper.py:105]: following the logic in engine `input_model_hash` is always called with `none` so the else part is never run. Is there a reason for this?

Comment (xiaoyu-work) [reply olive/engine/cloud_cache_helper.py:105]: Only the first input model hash for first Pass is None. The following pass's input model hash will be the output model hash from previous pass run returns in the for loop.

File: olive/engine/engine.py (hunk 1 above)
Comment (jambayk) [reply olive/engine/engine.py:808]: the inconsistency between `HfModel` and `HFModel` in this docstring still exists. My intention was to use `HfModel` as the casing but I missed updating here https://github.com/microsoft/Olive/blob/9e2c817b126f38d462680d72b20def711b26c668/olive/model/handler/hf.py#L27

Comment (xiaoyu-work) [reply olive/engine/engine.py:808]: Will update to HfModel.

**Review by jambayk — APPROVED** [review]

**Pull request merged**